brain/neuron/morphologyImpl.cpp

```diff
@@ -36,7 +36,7 @@ namespace neuron
 namespace
 {
 template< typename T >
-void _serializeArray( uint8_t*& dst,
+void _serializeArray( unsigned char*& dst,
```

Comment:
I read in SO that uin8_t is less safe than unsigned char. In some architectures uin8_t could be an extended numerical type for efficiency (the standard allows it) while unsigned char is always guaranteed to be the minimum addressable information unit (I wonder if it can be different from a byte nowadays). Also strict aliasing rules are different for unsigned char*, unsigned char* can alias any other type for uin8_t* the normal strict alias rules apply.

On the other hand I've just remembered and double checked that we are using -fno-strict-aliasing. We should remove that option and fix the code that doesn't work because the performance impact can be non-negligible, and we are not using __restrict__ pervasively to mitigate it.